client/my-sites/patterns/patterns-ssr.tsx

```diff
@@ -0,0 +1,34 @@
+import { useLocale } from '@automattic/i18n-utils';
+import classNames from 'classnames';
+import DocumentHead from 'calypso/components/data/document-head';
+import Main from 'calypso/components/main';
+import { usePatterns } from 'calypso/my-sites/patterns/hooks/use-patterns';
+
+import './style.scss';
+
+type Props = {
+	category: string;
+	isGridView?: boolean;
+};
+
+export default function PatternsSSR( { category, isGridView }: Props ) {
+	const locale = useLocale();
+	const { data: patterns } = usePatterns( locale, category );
+
+	return (
+		<Main isLoggedOut fullWidthLayout>
+			<DocumentHead title="WordPress Patterns" />
+
+			<h1>Build your perfect site with patterns</h1>
+
+			<div className={ classNames( 'patterns', { patterns_grid: isGridView } ) }>
+				{ patterns?.map( ( pattern ) => (
+					<div className="patterns__item patterns__item_loading" key={ pattern.ID }>
+						<div className="patterns__preview">Loading…</div>
```

Comment:
Just curious about how the loading state would look.

In the Assembler, we show a blank placeholder by passing the `minHeight` prop to `PatternRenderer`. That height is only used during the loading state while the pattern content height is being calculated. See https://github.com/Automattic/wp-calypso/blob/e5fe369ad0a544caf2eabba93a9a7d52b5fd15af/client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/pattern-list-renderer.tsx#L84 and https://github.com/Automattic/wp-calypso/blob/e5fe369ad0a544caf2eabba93a9a7d52b5fd15af/packages/block-renderer/src/components/block-renderer-container.tsx#L103

I noticed that the grid layout loading state is minimal compared to the full-width layout:

|GRID LAYOUT|FULL-WIDTH LAYOUT|
|-|-|
|<video src="https://github.com/Automattic/wp-calypso/assets/1881481/b2983356-db3e-4ff9-9075-5fd04b56f570">|<video src="https://github.com/Automattic/wp-calypso/assets/1881481/a153016c-4ae5-4acb-923f-3d1bed97c919">|

Reply:
Good call about looking closer at the loading state! I've made two changes to improve the loading state:
1. Pass a `placeholder` prop to `BlockRendererProvider`. This is apparently what we render while `BlockRendererProvider` is fetching data. Without it, we see a pretty jarring in-between state.
2. Pass a calculated `minHeight` prop to `PatternRenderer` that matches the initial `aspect-ratio`-based height.